docs/control-structures.rst

```diff
@@ -51,40 +56,45 @@ write::
 
 The names of output parameters can be omitted.
 The output values can also be specified using ``return`` statements.
-The ``return`` statements are also capable of returning multiple
-values, see :ref:`multi-return`.
+
+The ``return`` statements are also capable of returning multiple values using ``return (v0, v1, ..., vn)``. The number of parameters must equal the number of output parameters.
+
 Return parameters are initialized to zero; if they are not explicitly
 set, they stay to be zero.
 
 Input parameters and output parameters can be used as expressions in
 the function body.  There, they are also usable in the left-hand side
 of assignment.
 
+...note::
+    You can return all types from Solidity functions, but for structs, only from ``internal`` function calls. You can return ``memory`` structs from any function if you enable the new experimental ``ABIEncoderV2`` feature by adding ``pragma experimental ABIEncoderV2;`` to your source file. This feature introduces other experimental changes that you can read more about in the ABI_ guide.
```

Comment:
I am not sure if these are the only restrictions. Perhaps better to say something like `Some types currently cannot be returned from non-internal functions. If you enable ... then this activates more types, but very special types are limited to live inside a single contract only and cannot be transferred (for example mappings).` or something like that.

Reply:
@chriseth Actually, I think we are covering a lot of this better in the new WIP PR https://github.com/ethereum/solidity/pull/4615

Should we make the changes you suggest now, and then merge that PR later, or wait for the PR to take on-board all this feedback?

Reply:
Whatever makes the PRs smaller.

Reply:
@chriseth OK, updated with your suggestion and we'll go into more details on the other PR (and the other PR that hasn't yet begun).